v0/destinations/hs/util.js

```diff
@@ -238,26 +238,14 @@ const searchContacts = async (message, destination) => {
     }
   } else {
     // look for propertyName (key name) in traits
-    // Config.lookupField -> lookupField
-    // traits: { lookupField: email }
-    propertyName = traits[`${Config.lookupField}`];
+    // Config.lookupField -> firstname
+    // traits: { firstname: test name }
+    propertyName = Config.lookupField;
   }
-
-  // extract its value from the known propertyName (key name)
-  // if not found in our structure then look for it in traits
-  // Config.lookupField -> lookupField
-  // eg: traits: { lookupField: email, email: "test@test.com" }
   const value =
     getFieldValueFromMessage(message, propertyName) ||
     traits[`${propertyName}`];
 
-  if (!value) {
-    throw new CustomError(
-      `[HS] Identify:: '${propertyName}' lookup field for contact lookup not found in traits`,
-      400
-    );
-  }
-
```

Comment:
If value is not present how would we handle?

Reply:
Why not use this approach
```
function getLookupFieldValue(message, lookupField) {
      SOURCE_KEYS = ["traits", "context.traits", "properties"]
      value = getValueFromMessage(message, lookUpField);
      if (!value) {
        // Check in free-flowing object level
        SOURCE_KEYS.some(sourceKey => {
          value = getMappingFieldValueFormMessage(
            message,
            sourceKey,
            lookUpField
          );
          if (value) {
            return true;
          }
          return false;
        });
      }
   return value;
}
     
value =  getLookupFieldValue(message, lookupField) || getLookupFieldValue(message, "email");
if(!value) {
  error out
}


```

Reply:
addressed